### PR TITLE
IPOC-1141: migrate to IL Atlas

### PR DESCRIPTION
**Pull request by a-le-jan-dro**
**JIRA:** [IPOC-1141](https://clever.atlassian.net/browse/IPOC-1141)

**Overview:**

The IL managed Mongo DB has been done away with even sooner than I expected, so all jobs pulling from that have started failing and we must migrate. This necessitates more changes than expected due to the managed db not requiring authentication and this worker not supporting it. The database URL is moving from the payload to the config.

Associated ark-config PR: https://github.com/Clever/ark-config/pull/1764

**Testing:** Updated the tests, plan on testing in dev.

**Roll Out:**

- [ ] Updated config file path and collections in cron-admin jobs (as needed)

### CODE REVIEW

**Comment by bstein-clever** [comment]
LGTM as wel.

**Comment by a-le-jan-dro** [comment]
My apologies, this is now actually working, I believe (tested in `clever-dev`).

**Comment by bstein-clever** [comment]
If this needs a fresh review, can we clean up the history and wipe out anything that's no longer needed?
I see a "revert to master" - I'm guessing perhaps everything before that is irrelevant?

**Comment by a-le-jan-dro** [comment]
Okay this should be good to go now! I tested with every config in dev 🙏 Gonna update the production ark-config now. Also rebased so commit-by-commit review should be 👌 now.

File: main.go [hunk 1]
```diff
@@ -289,6 +281,13 @@ func main() {
 		log.Fatal(err)
 	}
 
+	mongoClient, err := mongoConnection(configYaml.URL, configYaml.User, configYaml.Password)
+	if err != nil {
```
Comment (bstein-clever) [inline main.go]: Isn't this error check the wrong way?

File: main.go [hunk 2]
```diff
@@ -283,10 +283,9 @@ func main() {
 
 	mongoClient, err := mongoConnection(configYaml.URL, configYaml.User, configYaml.Password)
 	if err != nil {
-		log.Println("Connected to mongo")
```
Comment (bstein-clever) [inline main.go]: Ahh, I see you fixed it. It might be worth consolidating those commits ...

**Comment by bstein-clever** [comment]
A couple of minor details, otherwise this looks good.

**Pull request merged**